.github/workflows/e2e.yaml

```diff
@@ -1,17 +1,24 @@
 name: Integration Testing
 
 on:
-  push:
-    branches: [main]
-  pull_request:
-    branches: [main]
-  workflow_dispatch:
+  workflow_run:
+    workflows: ["Build"]
+    types: 
+      - completed
```

Comment:
You might want to run the integration tests only if the build workflow completed successfully: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-a-workflow-based-on-the-conclusion-of-another-workflow

Reply:
Ack, will give the whole a try